src/compiler/parser.ts

```diff
@@ -594,7 +599,7 @@ namespace ts {
             // If this is a javascript file, proactively see if we can get JSDoc comments for
             // relevant nodes in the file.  We'll use these to provide typing informaion if they're
             // available.
-            if (isJavaScript(fileName)) {
+            if (isSourceFileJavaScript(sourceFile)) {
                 addJSDocComments();

```

Comment:
I don't think that's a deal breaker. Other things such as parent pointers, symbols, and locals aren't available until after binding. I think it is hard to justify the performance hit of a full AST walk just to have the JS doc information available before binding (which isn't actually needed by any of our tools).